File: VB/PrismSupportSample/DockManagerAdapter.vb
Imports System.Windows.Controls
Imports DevExpress.Xpf.Docking
Imports Prism.Regions

Namespace PrismSupportSample
    Public Class DockManagerAdapter
        Inherits RegionAdapterBase(Of DockLayoutManager)
        Public Sub New(ByVal BehaviorFactory As IRegionBehaviorFactory)
            MyBase.New(BehaviorFactory)
        End Sub
        Protected Overrides Function CreateRegion() As IRegion
            Return New SingleActiveRegion()
        End Function
        Protected Overrides Sub Adapt(ByVal region As IRegion, ByVal regionTarget As DockLayoutManager)
            Dim items() As BaseLayoutItem = regionTarget.GetItems()
            For Each item As BaseLayoutItem In items
                Dim regionName As String = RegionManager.GetRegionName(item)
                If (Not String.IsNullOrEmpty(regionName)) Then
                    Dim panel As LayoutPanel = TryCast(item, LayoutPanel)
                    If panel IsNot Nothing AndAlso panel.Content Is Nothing Then
                        Dim control As New ContentControl()
                        RegionManager.SetRegionName(control, regionName)
                        panel.Content = control
                    End If
                End If
            Next item
        End Sub
    End Class
End Namespace

File: CS/PrismSupportSample/DockManagerAdapter.cs
using System.Windows.Controls;
using DevExpress.Xpf.Docking;
using Prism.Regions;

namespace PrismSupportSample {
    public class DockManagerAdapter : RegionAdapterBase<DockLayoutManager> {
        public DockManagerAdapter(IRegionBehaviorFactory BehaviorFactory)
            : base(BehaviorFactory) {
        }
        protected override IRegion CreateRegion() {
            return new SingleActiveRegion();
        }
        protected override void Adapt(IRegion region, DockLayoutManager regionTarget) {
            BaseLayoutItem[] items = regionTarget.GetItems();
            foreach(BaseLayoutItem item in items) {
                string regionName = RegionManager.GetRegionName(item);
                if(!string.IsNullOrEmpty(regionName)) {
                    LayoutPanel panel = item as LayoutPanel;
                    if(panel != null && panel.Content == null) {
                        ContentControl control = new ContentControl();
                        RegionManager.SetRegionName(control, regionName);
                        panel.Content = control;
                    }
                }
            }
        }
    }
}
